filebeat/input/udp/input.go

```diff
@@ -116,10 +152,73 @@ func (p *Input) Stop() {
 
 	logp.Info("Stopping UDP input")
 	p.udp.Stop()
+	p.metrics.close()
 	p.started = false
 }
 
 // Wait suspends the UDP input
 func (p *Input) Wait() {
 	p.Stop()
 }
+
+// inputMetrics handles the input's metric reporting.
+type inputMetrics struct {
+	unregister func()
+
+	lastPacket time.Time
+
+	device         *monitoring.String // name of the device being monitored
+	packets        *monitoring.Uint   // number of packets processed
+	bytes          *monitoring.Uint   // number of bytes processed
+	bufferLen      *monitoring.Uint   // configured read buffer length
+	arrivalPeriod  metrics.Sample     // histogram of the elapsed time between packet arrivals
+	processingTime metrics.Sample     // histogram of the elapsed time between packet receipt and publication
+}
+
+// newInputMetrics returns an input metric for the UDP processor. If id is empty
+// a nil inputMetric is returned.
+func newInputMetrics(id, device string, buflen uint64) *inputMetrics {
+	if id == "" {
+		return nil
+	}
+	reg, unreg := inputmon.NewInputRegistry("udp", id+"::"+device, nil)
+	out := &inputMetrics{
+		unregister:     unreg,
+		bufferLen:      monitoring.NewUint(reg, "udp_read_buffer_length"),
+		device:         monitoring.NewString(reg, "device"),
+		packets:        monitoring.NewUint(reg, "udp_packets"),
+		bytes:          monitoring.NewUint(reg, "udp_bytes"),
+		arrivalPeriod:  metrics.NewUniformSample(1024),
+		processingTime: metrics.NewUniformSample(1024),
+	}
+	_ = adapter.NewGoMetrics(reg, "udp_arrival_period", adapter.Accept).
+		Register("histogram", metrics.NewHistogram(out.arrivalPeriod))
+	_ = adapter.NewGoMetrics(reg, "udp_processing_time", adapter.Accept).
+		Register("histogram", metrics.NewHistogram(out.processingTime))
+
```

Comment:
Let me propose some slightly more generic names. Before we ship this in the next release I plan to look across the inputs to see if there are opportunities for alignment on naming.

- `received_packets_total`
- `received_events_total` - My thinking is that s/packet/events/ will be more generic and applicable across more inputs.
- `arrival_period`
- `processing_time`

Reply:
> - `received_packets_total`
> - `received_events_total` - My thinking is that s/packet/events/ will be more generic and applicable across more 

Is the first of these intended to be `received_bytes_total`?

Reply:
Yep. That should have been received_bytes_total.